lib/src/components/NxAccordion/__tests__/NxAccordion.test.tsx

```diff
@@ -5,277 +5,221 @@
-      expect(wrapper).toHaveClassName('nx-accordion__summary-wrapper');
-      expect(wrapper).toContainReact(<span>Foo</span>);
-    });
+      const { rerender, container } = quickRender();
 
-    it('sets the nx-accordion__header class along with any provided className on the header', function() {
-      const component = getMountedComponent({
-            className: 'accordion-class',
-            children: (
-              <NxAccordion.Header className="header-class">
-                <span>Foo</span>
-              </NxAccordion.Header>
-            )
-          }),
-          header = component.find('summary');
-
-      expect(header).toHaveClassName('nx-accordion__header');
-      expect(header).toHaveClassName('header-class');
-      expect(getMountedComponent({ children: <NxAccordion.Header /> }).find('summary'))
-          .toHaveClassName('nx-accordion__header');
-    });
+      rerender(
+        <NxAccordion.Header id="headerId">
+          <span>Foo</span>
+        </NxAccordion.Header>
+      );
 
-    it('renders an icon with nx-accordion__chevron class as the first child of the header wrapper', function() {
-      const component = getMountedComponent({
-            children: (
-              <NxAccordion.Header>
-                <span>Foo</span>
-                <div className="nx-btn-bar">
-                  <NxButton />
-                </div>
-              </NxAccordion.Header>
-            )
-          }),
-          wrapper = component.find('summary').children();
-
-      expect(wrapper.children().first()).toMatchSelector(NxFontAwesomeIcon);
-      expect(wrapper.children().at(1)).toMatchSelector('span');
-      expect(wrapper.children().last()).toMatchSelector('div');
-
-      expect(wrapper.find(NxFontAwesomeIcon)).toHaveClassName('nx-accordion__chevron');
+      expect(container.querySelector('#headerId')).toBeInTheDocument();
+      expect(container.querySelector('.nx-accordion__summary-wrapper')).toBeInTheDocument();
     });
 
-    it('uses faChevronCircleDown as the icon when the accordion is closed', function() {
-      const component = getMountedComponent({
-            children: (
-              <NxAccordion.Header>
-                <span>Foo</span>
-              </NxAccordion.Header>
-            )
-          }),
-          header = component.find('summary');
-
-      expect(header.find(NxFontAwesomeIcon)).toHaveProp('icon', faChevronCircleDown);
-
-      const explicitOpenComponent = getMountedComponent({
-            open: false,
-            children: (
-              <NxAccordion.Header>
-                <span>Foo</span>
-              </NxAccordion.Header>
-            )
-          }),
-          explicitOpenHeader = explicitOpenComponent.find('summary');
-
-      expect(explicitOpenHeader.find(NxFontAwesomeIcon)).toHaveProp('icon', faChevronCircleDown);
+    it('sets the nx-accordion__header class along with any provided className on the header', function() {
```

Comment:
`.nx-accordion__header` is an impl detail.  I'd replace this with our usual style of test for custom classnames that checks that all classes present without a custom classname are also still present with a custom classname.

Reply:
> I think the custom classnames test is already added:

If I understand what you're linking to correctly, that's the classname test for `NxAccordion`.  You also need one for `NxAccordion.Header`.

Reply:
> Added here: https://github.com/sonatype/sonatype-react-shared-components/commit/6bab09f43a65a7fe9dee14d24d331842795c88d2

The test needs to add the `foo` class to the header, not the accordion.  And the comparison with the default classes needs to be done across the header elements, not the accordion elements.

Reply:
> > Added here: [6bab09f](https://github.com/sonatype/sonatype-react-shared-components/commit/6bab09f43a65a7fe9dee14d24d331842795c88d2)
> 
> The test needs to add the `foo` class to the header, not the accordion. And the comparison with the default classes needs to be done across the header elements, not the accordion elements.

Updated here: 254a5eecf167c8f59dd9b9a0d73a5aa91cc15a31